cmd/notation/sign.go

```diff
@@ -66,10 +76,16 @@ Example - Sign an OCI artifact stored in a registry and specify the signature ex
 	opts.SecureFlagOpts.ApplyFlags(command.Flags())
 	cmd.SetPflagExpiry(command.Flags(), &opts.expiry)
 	cmd.SetPflagPluginConfig(command.Flags(), &opts.pluginConfig)
+	command.Flags().StringVar(&opts.imageSpec, "image-spec", artifactManifest, "manifest type for signatures. options: v1.1-artifact, v1.1-image")
 	return command
 }
 
 func runSign(command *cobra.Command, cmdOpts *signOpts) error {
+	// sanity check
+	if !validateImageSpec(cmdOpts.imageSpec) {
+		return fmt.Errorf("image spec has to be in %v, got %s", supportedImageSpec, cmdOpts.imageSpec)
+	}
+
```

Comment:
Can move the section out of runSign() to make the function clear. What about move it to

```go
		RunE: func(cmd *cobra.Command, args []string) error {
            // Here
			return runSign(cmd, opts)
		},
```